src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -15,49 +14,32 @@ use versionize::{VersionMap, Versionize, VersionizeResult};
 use versionize_derive::Versionize;
 use vm_memory::{Address, GuestAddress, GuestMemoryMmap};
 
+use super::VcpuConfig;
+use crate::arm_templates::g3g2::g3g2;
+use crate::arm_templates::{apply_template, Error as TemplateError};
+use crate::vmm_config::machine_config::CpuFeaturesTemplate;
 use crate::vstate::vcpu::VcpuEmulation;
 use crate::vstate::vm::Vm;
 
 /// Errors associated with the wrappers over KVM ioctls.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum Error {
-    /// Error configuring the general purpose aarch64 registers.
+    #[error("Error configuring the general purpose aarch64 registers: {0}")]
     ConfigureRegisters(arch::aarch64::regs::Error),
-    /// Cannot open the kvm related file descriptor.
+    #[error("Cannot open the kvm related file descriptor: {0}")]
     CreateFd(kvm_ioctls::Error),
-    /// Error getting the Vcpu preferred target on Arm.
+    #[error("Error getting the Vcpu preferred target on Arm: {0}")]
     GetPreferredTarget(kvm_ioctls::Error),
-    /// Error doing Vcpu Init on Arm.
+    #[error("Error doing Vcpu Init on Arm: {0}")]
     Init(kvm_ioctls::Error),
-    /// Failed to set value for some arm specific register.
+    #[error("Error applying template: {0}")]
+    CpuTemplate(TemplateError),
+    #[error("Failed to set value for some arm specific register: {0}")]
```

Comment:
nit: could be an opportunity to unify the error message language (aarch64 vs Arm vs arm).

Reply:
it seems all modules named `aarch64` so I can rename everything to use this naming style